### ISSUE TITLE
How to build Brackets on openSUSE from source code

### ISSUE THREAD

**Issue by fusion809** [adobe/brackets#12228]
Hi,

Brackets has official Debian packages for Debian and Ubuntu, but no RPM packages for Fedora or openSUSE, hence why I am here to ask how one might install Brackets from source code. I found no instructions on this in this repository's README and I've run the most straightforward drill to install a Node.js-based program (`npm install` from its top-level directory) and then I ran `npm start` to start it and it returned:

``` bash
npm ERR! Linux 4.4.0-3-default
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "start"
npm ERR! node v5.4.1
npm ERR! npm  v3.7.2

npm ERR! missing script: start
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /home/fusion809/Programs/brackets/npm-debug.log
```

So, hence why I am here and why I would appreciate it if someone could tell me how to. 

Thanks for your time,
Brenton

**Comment by fusion809** [comment]
Oops, I just found out how to do this myself by reading the [Wiki article on building](https://github.com/adobe/brackets/wiki/Building-Brackets-Releases). Might I suggest that this guide be linked to in this repo's README?

**Comment by fusion809** [comment]
I tried those instructions, namely running `grunt build` after `npm install` and now I am getting this error:
```bash
Running "eslint:src" (eslint) task


Running "jasmine:src" (jasmine) task
Testing jasmine specs via phantom
>> Error: Script error
>> http://requirejs.org/docs/errors.html#scripterror at
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> .grunt/grunt-contrib-jasmine/require.js:14 z
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> Error: Script error
>> http://requirejs.org/docs/errors.html#scripterror at
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> .grunt/grunt-contrib-jasmine/require.js:14 z
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> Error: Script error
>> http://requirejs.org/docs/errors.html#scripterror at
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> .grunt/grunt-contrib-jasmine/require.js:14 z
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> Error: Load timeout for modules: text!language/languages.json_unnormalized2,i18n!nls/strings_unnormalized3,i18n!nls/urls_unnormalized4,i18n!nls/strings-app_unnormalized5,text!config.json_unnormalized6                                                                                                                                             
>> http://requirejs.org/docs/errors.html#timeout at
>> .grunt/grunt-contrib-jasmine/require.js:31 
>> .grunt/grunt-contrib-jasmine/require.js:14 z
>> .grunt/grunt-contrib-jasmine/require.js:16 w
>> .grunt/grunt-contrib-jasmine/require.js:16 

Warning: PhantomJS timed out, possibly due to an unfinished async spec. Use --force to continue.

Aborted due to warnings.
```

**Comment by vishwajithsandaru** [comment]
i also got the above error, its appreciable if somebody can help

**Comment by ficristo** [comment]
My first impression is that `phantomjs` didn't download correctly.
Could you try running `npm install` again? Maybe try to delete `node_modules` folder first too.

**Comment by ficristo** [comment]
I wasn't able to reproduce on Ubuntu16.04.
I close this since we didn't hear back from you. If you still hit this problem please reopen after answering my question.

**Comment by kuboosoft** [comment]
In Fedora we have the same problem...

```
PhantomJS not found on PATH
Download already available at /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2
Verified checksum of previously downloaded file
Extracting tar contents (via spawned process)
Removing /builddir/build/BUILD/brackets-86b5e2d/node_modules/phantomjs-prebuilt/lib/phantom
Copying extracted folder /tmp/phantomjs/phantomjs-2.1.1-linux-x86_64.tar.bz2-extract-1506549255084/phantomjs-2.1.1-linux-x86_64 -> /builddir/build/BUILD/brackets-86b5e2d/node_modules/phantomjs-prebuilt/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /builddir/build/BUILD/brackets-86b5e2d/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs
Brackets@1.12.0-0 /builddir/build/BUILD/brackets-86b5e2d
+-- abbrev@1.0.9 
+-- aproba@1.0.4 
+-- asap@2.0.5 
+-- config-chain@1.1.11 
+-- fstream@1.0.10 
+-- fstream-npm@1.2.0 
| `-- fstream-ignore@1.0.5
|   `-- minimatch@3.0.3 
+-- glob@7.1.0 
| `-- minimatch@3.0.3 
+-- graceful-fs@4.1.9 
+-- has-unicode@2.0.1 
+-- inherits@2.0.3 
+-- lockfile@1.0.2 
+-- lodash.clonedeep@4.5.0 
+-- lodash.union@4.6.0 
+-- lodash.uniq@4.5.0 
+-- lodash.without@4.4.0 
+-- minimatch@3.0.3 
+-- node-gyp@3.4.0 
| `-- minimatch@3.0.3 
+-- npm@3.10.10
| +-- init-package-json@1.9.4
| | `-- glob@6.0.4
| |   `-- minimatch@3.0.3 
| +-- request@2.75.0 
| `-- semver@5.3.0 
+-- npm-package-arg@4.2.0 
+-- npm-registry-client@7.2.1 
+-- npm-user-validate@0.1.5 
+-- npmlog@4.0.0 
+-- once@1.4.0 
+-- opener@1.4.2 
+-- path-is-inside@1.0.2 
+-- phantomjs-prebuilt@2.1.15  extraneous
+-- read-package-tree@5.1.5 
+-- readable-stream@2.1.5 
+-- retry@0.10.0 
+-- rimraf@2.5.4 
+-- sorted-object@2.0.1 
+-- which@1.2.11 
+-- wrappy@1.0.2 
`-- write-file-atomic@1.2.0 

npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.1.2
npm WARN optional Skipping failed optional dependency /watchpack/chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.1.2
npm WARN Brackets@1.12.0-0 No license field.
+ export PATH=/builddir/nvm/versions/node/v6.0.0/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin:/builddir/build/BUILD/brackets-86b5e2d/node_modules/.bin/
+ PATH=/builddir/nvm/versions/node/v6.0.0/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin:/builddir/build/BUILD/brackets-86b5e2d/node_modules/.bin/
+ ./node_modules/.bin/grunt build --platforms=linux64
Running "write-config:dist" (write-config) task

Running "eslint:src" (eslint) task

Running "jasmine:src" (jasmine) task
Testing jasmine specs via phantom

Running PhantomJS...ERROR
>> PhantomJS has crashed. Please read the crash reporting guide at https://github.com/ariya/phantomjs/wiki/Crash-Reporting and file a bug report at https://github.com/ariya/phantomjs/issues/new with the crash dump file attached: /tmp/15ef0d97-f85a-0e43-68130019-69ad37ed.dmp 0 [ 'PhantomJS has crashed. Please read the crash reporting guide at https://github.com/ariya/phantomjs/wiki/Crash-Reporting and file a bug report at https://github.com/ariya/phantomjs/issues/new with the crash dump file attached: /tmp/15ef0d97-f85a-0e43-68130019-69ad37ed.dmp' ]
Warning: PhantomJS exited unexpectedly with exit code null. Use --force to continue.

Aborted due to warnings.
```

**Comment by Mis012** [comment]
maybe try to reproduce it on openSUSE then? still a problem here
